File: ios/RCTMGL-v10/MBXMarkerViewManager.m
#import "React/RCTBridgeModule.h"
#import <React/RCTViewManager.h>
#import <Foundation/Foundation.h>

@interface RCT_EXTERN_REMAP_MODULE(MBXMarkerView, MBXMarkerViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(coordinate, NSString)
RCT_EXPORT_VIEW_PROPERTY(anchor, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(allowOverlap, BOOL)
RCT_EXPORT_VIEW_PROPERTY(isSelected, BOOL)

@end

File: ios/RCTMGL-v10/MBXMarkerViewManager.swift
import Foundation
import MapboxMaps

@objc(MBXMarkerViewManager)
class MBXMarkerViewManager : RCTViewManager {
  @objc
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  override func view() -> UIView! {
    return MBXMarkerView()
  }
}
